app/models/user.rb

```diff
@@ -1,12 +1,14 @@
 class User < ActiveRecord::Base
+  validates_presence_of :email, :password, :full_name
+  validates_uniqueness_of :email
+
   has_secure_password validations: false
   has_many :queue_items, -> {order("position")}
   has_many :reviews, -> {order("created_at DESC")}
   has_many :following_relationships, class_name: 'Relationship', foreign_key: :follower_id
   # should return all the relationships where the user is the follower
 
-  validates_presence_of :email, :password, :full_name
-  validates_uniqueness_of :email
+  before_create :generate_token
```

Comment:
The problem of this is that the token serves in fact as password to users account. Whoever knows the token can change the original password. That means you should either encrypt the password when it is in the db since users creation or basically permanently present (other token after refresh).

Ideally, you should create the token only upon request. And expire it after some time.